db/migrate/20240328075919_add_trial_ended_at_to_subscriptions.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+class AddTrialEndedAtToSubscriptions < ActiveRecord::Migration[7.0]
+  class Subscription < ApplicationRecord
+    belongs_to :customer
+    belongs_to :plan
+
+    # NOTE: We reimplement the logic from Subscription#initial_started_at  differently to avoid N+1 queries
+    #       eager loading subscription.customer.subscriptions is not enough.
+    def initial_started_at
+      customer.subscriptions.select do |s|
+        s.external_id == external_id && s.started_at.present?
+      end.sort_by(&:started_at).first&.started_at || subscription_at
+    end
+  end
+
+  class Customer < ApplicationRecord
+    has_many :subscriptions
+  end
+
+  class Plan < ApplicationRecord
+    has_many :subscriptions
+  end
+
+  def up
+    add_column :subscriptions, :trial_ended_at, :datetime
+
+    Subscription
+      .joins(:plan)
+      .where(trial_ended_at: nil)
+      .where.not(plans: { trial_period: nil })
+      .includes(:plan, customer: :subscriptions)
+      .find_each do |subscription|
+
+      trial_ended_at = subscription.initial_started_at.to_date + subscription.plan.trial_period.days
+
+      next if trial_ended_at.to_date >= Date.today
```

Comment:
Rails/Date: Do not use `Date.today` without zone. Use `Time.zone.today` instead.